### ISSUE TITLE
Automated upload of existing OBEU-datasets on the server & staging triplestore

### ISSUE THREAD

**Issue by schmaluk** [openbudgets/platform#13]
We Need to upload our existing OBEU-datasets:
https://github.com/openbudgets/datasets
to our Server in an automated way (ideally whenever the datasets are updated in the github-repository).
They should become available as plain file here:
http://eis-openbudgets.iais.fraunhofer.de/dumps/
and also in the staging triplestore in a named graph.

**Comment by jindrichmynarz** [comment]
Instead of syncing the dataset dumps, we should import the pipelines from https://github.com/openbudgets/datasets and re-create the datasets on the OpenBudgets.eu server.

Doing this also tests that each dataset is reproducible given the provided pipeline. If the imported pipelines don't work, please raise an issue on the datasets repository.

**Comment by schmaluk** [comment]
Thanks @jindrichmynarz. Will try to do that. Then this should probably not be automated, if this needs a human eye to supervise.

**Comment by jindrichmynarz** [comment]
Maybe I misunderstand you, but the pipelines themselves are what automate the ETL processes. Do you want to automate also the pipeline import?

**Comment by schmaluk** [comment]
Yes, I thought initially that was a good idea for reducing work. Whenever sthg is pushed to github. But now I think it is probably better to do it manually for now for validation.

**Comment by jindrichmynarz** [comment]
Theoretically, you could retrieve the pipelines in `*.jsonld` files using the GitHub's API and import them to LP-ETL via its API.

**Comment by jakubklimek** [comment]
I think we should first define the intended workflow as I think this is the other way around. First a pipeline producing a dataset is developed locally, then it should be deployed to OBEU server LP-ETL instance (manually, to adjust/check output directories and triplestore settings) where it produces the data and stores it. Then this pipeline and possibly its input and output (if not too large) is stored in GitHub. GitHub therefore serves as an archive, not as an input.

**Comment by jindrichmynarz** [comment]
I think GitHub should serve us primarily as means to sharing pipelines. ETL developers should implement pipelines locally, push the pipelines to GitHub, out of which they should be imported to the staging server and replicated.

**Comment by schmaluk** [comment]
My first thought was that just the produced datasets can be grabbed on github and imported (maybe even without LP). This could get fully automated. But then the Server/user does not check, if the Pipelines are working correctly...
1) If you use LP on the OBEU-server in the first place and github as Archive, that is probably more responsive for the dataset/etl developer, since he can check the datasets also in the triplestores immediately. But errors get detected by other people more likely afterwards when they are shared to the public on github. Also all dataset/etl developers Need to get ssh Access for this workflow.
2) With the other approach they can be checked by others before. And after being approved by other ppl by reviewing on github, can be imported to the Server manually (by me). If this is done by me however, it might be even less responsive, since Im not an expert on datasets and Need to raise issues, when I detect them.

I am neutral. We can do, whatever works best for you.

**Comment by jindrichmynarz** [comment]
I think we should adopt a workflow similar to your option 2: two people should be involved. The pipelines should be made reusable without much knowledge about their internals. Necessary configuration or prerequisites should be documented.

**Comment by jakubklimek** [comment]
Option 2 makes sense if:
1. There will be a substantial amount of new datasets, for which new manual pipelines will be created. Is this planned? T2.1 and T2.2 have already ended, so this would have to be only to support analytics and data mining in the next 10 months on top of datasets, that will not come from OpenSpending. Otherwise it is not worth it to establish a proper workflow.
2. There will be a proper validation workflow established, with clear definition of who is responsible for the review of the pipeline itself, and the involved steps, similarly to the Deliverable workflow. Since even some of the project Deliverables did not follow the workflow, I don't see it as realistic to have such workflow for the manually created pipelines (I assume there will be more datasets than deliverables). Typically as it is now, a developer creates the pipeline, uploads it and the data and that's it. Maybe someone notices problems in the resulting data and notifies the developer to fix it. But in reality, will pipeline developers (i.e. UEP, OKFGR, UBONN, IAIS) import pipelines made by others to validate them? Or is it really just for archivation? This makes sense for reusable pipeline fragments, but I am not sure about the resulting pipelines.
3. There is a process for injecting credentials into the pipelines, as the credentials will not be stored in the GitHub version of the pipelines. This means some restrictions on the shape of the pipelines. This would not be necessary if the developer would be responsible for adjusting the pipeline before loading it manually to the OBEU instance.

To sum up, if we think there will be at least tens of new manually created datasets for data analytics and data mining during the next 10 months and we want to add the validation workflow for them, option 2 is worth investigating. Otherwise I would stick with option 1.

**Comment by schmaluk** [comment]
So do we agree then that each ETL/datasets-Pipeline-developer will upload their pipelines themselves to the OBEU-server or am I wrong? However I can assist/take over, if it just means to upload the pipelines from our git-repo & they are busy with other tasks.
I can provide (with Docker) some additional LinkedPipes instances for the transformations. In this way the FDP-to-RDF-pipeline doesnt get blocked by the transformations and the transformations doesnt block each other.
In any case the pipelines need to be adapted in order to:
 a) store the rdf-files on our server:
        - Use the Target File directory: /dumps (just a softlink to the "real directory" & can also be initialized as a git repo) where content is exposed @http://eis-openbudgets.iais.fraunhofer.de/dumps/
        - In case the Files-to-Scp-component is used, please ask me for the credentials for the SSH-upload-user.
 b) upload the datasets in a separate named graph to our staging triplestore:
        - Use the Graph store protocol endpoint: http://triple_store_staging:3030/fuseki/data
        - no credentials
It is basically the same config which is used in the FDP-to-RDF-pipeline on the server.
Afterwards the pipelines & datasets should be manually archived on github(?)

**Comment by jakubklimek** [comment]
I would agree. In addition, if this is not established elsewhere, I would like to establish metadata rules for datasets uploaded to OBEU servers (in coordination with @larjohn as discussed in https://github.com/openbudgets/datasets/issues/24), which would affect not only the manually created datasets, but also the automatically created ones and therefore the FDP-to-RDF pipeline (cc @marek-dudas). Specifically (open for discussion):
1. Metadata should be provided in a separate graph named <dataset-graph-uri/metadata>
2. In manually created pipelines, these will be created using the DCAT-AP Dataset and DCAT-AP Distribution components, in FDP-to-RDF these would be generated based on the input FDP.

Dataset metadata should contain (from https://github.com/openbudgets/datasets/issues/24):
- [ ] Organization: Publisher IRI, publisher name, publisher type
- [ ] Author of the dataset + uploader of the dataset: Actually, there is support for one contact point in DCAT-AP: Contact point type, name and email. Is there an example where we need 2?
- [ ] Dataset IRI, name and description
- [ ] Fiscal period: Temporal coverage start, Temporal coverage end.

In addition, there is Distribution metadata, that should contain:
- [ ] Distribution IRI: can be generated automatically by the component
- [ ] Access URL and Download URL (the URL from which the dump can be downloaded after publishing)
- [ ] Media type and file format
- [ ] License and License type: This we have not established yet, but is a really important part if our data is to be in fact open data. License type should be IMHO some "Public domain" type.

The resulting metadata looks like this:

```
@prefix dcat: <http://www.w3.org/ns/dcat#> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix foaf: <http://xmlns.com/foaf/0.1/> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix schema: <http://schema.org/> .
@prefix vcard: <http://www.w3.org/2006/vcard/ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

<http://data.openbudgets.eu/resource/dataset/TEST> a dcat:Dataset ;
    dcterms:description "Testing dataset description"@en ;
    dcterms:modified "2016-07-22"^^xsd:date ;
    dcterms:publisher <http://www.vse.cz> ;
    dcterms:temporal <http://data.openbudgets.eu/resource/dataset/TEST/temporal> ;
    dcterms:title "Testing dataset"@en ;
    dcat:contactPoint <http://data.openbudgets.eu/resource/dataset/TEST/contactPoint> ;
    dcat:distribution <http://data.openbudgets.eu/resource/dataset/TEST/distribution> .

<http://data.openbudgets.eu/resource/dataset/TEST/contactPoint> a vcard:Individual,
        vcard:Kind ;
    vcard:fn "Jakub Klímek" ;
    vcard:hasEmail "klimek@opendata.cz" .

<http://data.openbudgets.eu/resource/dataset/TEST/distribution> a dcat:Distribution ;
    dcterms:format <http://publications.europa.eu/resource/authority/file-type/RDF_TRIG> ;
    dcterms:license <http://openbudgets.eu/resources/data-license> ;
    dcterms:modified "2016-07-22"^^xsd:date ;
    dcat:accessURL <http://eis-openbudgets.iais.fraunhofer.de/dumps/test.trig> ;
    dcat:downloadURL <http://eis-openbudgets.iais.fraunhofer.de/dumps/test.trig> ;
    dcat:mediaType <http://www.iana.org/assignments/media-types/application/trig> .

<http://data.openbudgets.eu/resource/dataset/TEST/temporal> a dcterms:PeriodOfTime ;
    schema:endDate "2016-07-31"^^xsd:date ;
    schema:startDate "2016-07-01"^^xsd:date .

<http://openbudgets.eu/resources/data-license> a dcterms:LicenseDocument ;
    dcterms:type <http://purl.org/adms/licencetype/PublicDomain> .

<http://publications.europa.eu/resource/authority/file-type/RDF_TRIG> a dcterms:MediaTypeOrExtent .

<http://purl.org/adms/licencetype/PublicDomain> a <http://www.w3.org/2004/02/skos/core#Concept> .

<http://www.iana.org/assignments/media-types/application/trig> a dcterms:MediaTypeOrExtent .

<http://www.vse.cz> a foaf:Agent ;
    dcterms:type <http://purl.org/adms/publishertype/Academia-ScientificOrganisation> ;
    foaf:name "Vysoká škola ekonomická v Praze"@cs,
        "University of Economics, Prague"@en .
```

and a pipeline fragment which generates this data is here:
[OBEU Dataset metadata demo.txt](https://github.com/openbudgets/platform/files/377837/OBEU.Dataset.metadata.demo.txt)

I recommend that each pipeline developer creates his/her own pipeline fragment for the metadata so that only the properties that change among datasets can be changed.

**Comment by jindrichmynarz** [comment]
I think metadata is another large issue that can be discussed separately.

Some points on the metadata issue:
- Link from a `qb:DataSet` to its metadata graph should be explicit (e.g., via `rdfs:seeAlso`), instead of based on convention, such as appending `/metadata` to the dataset's IRI. In this way, consumers may simply do things like `?dataset rdfs:seeAlso/dcterms:title ?title .` without needing to mash strings together in IRIs.
- Some distribution metadata, such as the download URL, depends on the configuration of loaders. This requires duplication in configuration, which is potentially error-prone. Configurable components might help, but may be too complex.

What is actually our current use case for having distribution metadata? I understand the use case for dataset metadata described by @larjohn, but I wonder what we need distribution metadata for.

**Comment by jakubklimek** [comment]
> Link from a qb:DataSet to its metadata graph should be explicit (e.g., via rdfs:seeAlso), instead of based on convention, such as appending /metadata to the dataset's IRI. In this way, consumers may simply do things like ?dataset rdfs:seeAlso/dcterms:title ?title . without needing to mash strings together in IRIs.

Sure, but still the metadata graph IRI should follow some convention. Actually, the rdfs:seeAlso should technically lead to the dcat:Dataset IRI (which should be the same as the dataset graph IRI), not the metadata graph IRI.

> Some distribution metadata, such as the download URL, depends on the configuration of loaders. This requires duplication in configuration, which is potentially error-prone. Configurable components might help, but may be too complex.

This also depends on the configuration of the publishing server. Sure, it is error-prone, on the other hand we don't expect many manually created pipelines and most of the datasets will be generated automatically. This is not a big issue.

> What is actually our current use case for having distribution metadata? I understand the use case for dataset metadata described by @larjohn, but I wonder what we need distribution metadata for.

One reason is being compliant with DCAT-AP, and along with it goes the possibility of generating a DCAT-AP compatible catalog of OBEU datasets. This is definitely worth the few clicks. 

On the other hand, the use of the components does not have to be mandatory. If it seems more convenient for yo, you can generate the metadata in any other way such as SPARQL Construct as long as it follows the specification.

**Comment by jindrichmynarz** [comment]
> Sure, but still the metadata graph IRI should follow some convention.

Agreed. Let's propose a convention in D1.5.

> Actually, the rdfs:seeAlso should technically lead to the dcat:Dataset IRI (which should be the same as the dataset graph IRI), not the metadata graph IRI.

I think we agreed that the instance of `qb:DataSet` also instantiates `dcat:Dataset` and also its IRI is used as the named graph IRI. Thus there would be no benefit in having `<dataset> rdfs:seeAlso <dataset>`. However, having an explicit link to the metadata graph would be useful.

I agree with the motivation to use DCAT distribution metadata.

**Comment by skarampatakis** [comment]
Hi, we are on the procedure to upload the Greek Municipalities datasets on Fuseki. Currently all these datasets use the old metadata component carried from UV. For instance we can see an example below:

```
@prefix dc: <http://purl.org/dc/elements/1.1/> .
@prefix foaf: <http://xmlns.com/foaf/0.1/> .
@prefix terms: <http://purl.org/dc/terms/> .
@prefix sesame: <http://www.openrdf.org/schema/sesame#> .
@prefix fn: <http://www.w3.org/2005/xpath-functions#> .

<http://data.openbudgets.eu/resource/dataset/budget-thessaloniki-expenditure-2011> terms:creator <http://okfn.gr/team/#okfn-bratsas> , <http://okfn.gr/team/#okfn-karampatakis> ;
    terms:issued "2016-01-22"^^xsd:date ;
    terms:language <http://id.loc.gov/vocabulary/iso639-1/el> ;
    terms:license <http://creativecommons.org/licenses/by-nc-nd/3.0/> ;
    terms:modified "2016-05-18"^^xsd:date ;
    terms:publisher <http://openbudgets.eu> ;
    terms:title "Προϋπολογισμός Εξόδων του Δήμου Θεσσαλονίκης (Ελλάδα) για το έτος 2011"@el ;
    a <http://www.w3.org/ns/dcat#Dataset> , qb:DataSet ;
    <http://www.w3.org/ns/dcat#contactPoint> <http://data.openbudgets.eu/resource/dataset/budget-thessaloniki-expenditure-2011/contactPoint> ;
    foaf:name "OpenBudgets.eu" ;
    obeu-attribute:currency obeu-currency:EUR ;
    obeu-dimension:fiscalYear <http://reference.data.gov.uk/id/year/2011> ;
    obeu-dimension:operationCharacter obeu-operation:expenditure ;
    obeu-dimension:organization <http://el.dbpedia.org/resource/Δήμος_Θεσσαλονίκης> ;
    dc:publisher <http://openbudgets.eu> ;
    terms:contributor _:node1aj2g43bfx4084 ;
    qb:structure <http://data.openbudgets.eu/ontology/dsd/budget-thessaloniki-expenditure-2011> ;
    rdfs:label "Municipality of Thessaloniki (Greece) expenditure Budget for the fiscal year 2011"@en ;
    qb:slice thess-be2011-slice:1 , thess-be2011-slice:10 , thess-be2011-slice:100 , thess-be2011-slice:1000 , thess-be2011-slice:1001 , thess-be2011-slice:1002 , thess-be2011-slice:1003 , thess-be2011-slice:1004 , thess-be2011-slice:1005 , thess-be2011-slice:1006 , thess-be2011-slice:1007 , thess-be2011-slice:1008 , thess-be2011-slice:1009 , thess-be2011-slice:101 , thess-be2011-slice:1010 , thess-be2011-slice:1011 , thess-be2011-slice:1012 , thess-be2011-slice:1013 , thess-be2011-slice:1014 , thess-be2011-slice:1015 , thess-be2011-slice:1016 , thess-be2011-slice:1017 , thess-be2011-slice:1018 , thess-be2011-slice:1019 , thess-be2011-slice:102 , thess-be2011-slice:1020 , thess-be2011-slice:1021 , thess-be2011-slice:1022 , thess-be2011-slice:1023 , thess-be2011-slice:1024 , thess-be2011-slice:1025 , thess-be2011-slice:1026 , thess-be2011-slice:1027 , thess-be2011-slice:1028 , thess-be2011-slice:1029 , thess-be2011-slice:103 , thess-be2011-slice:1030 , thess-be2011-slice:1031 , thess-be2011-slice:1032 , thess-be2011-slice:1033 , thess-be2011-slice:1034 , thess-be2011-slice:1035 , thess-be2011-slice:1036 , thess-be2011-slice:1037 , thess-be2011-slice:1038 , thess-be2011-slice:1039 , thess-be2011-slice:104 , thess-be2011-slice:1040 , thess-be2011-slice:1041 , thess-be2011-slice:1042 , thess-be2011-slice:1043 , thess-be2011-slice:1044 , thess-be2011-slice:1045 , thess-be2011-slice:1046 , thess-be2011-slice:1047 , thess-be2011-slice:1048 , thess-be2011-slice:1049 , thess-be2011-slice:105 , thess-be2011-slice:1050 , thess-be2011-slice:1051 , thess-be2011-slice:1052 , thess-be2011-slice:1053 , thess-be2011-slice:1054 , thess-be2011-slice:1055 , thess-be2011-slice:1056 , thess-be2011-slice:1057 , thess-be2011-slice:1058 , thess-be2011-slice:1059 , thess-be2011-slice:106 , thess-be2011-slice:1060 , thess-be2011-slice:1061 , thess-be2011-slice:1062 , thess-be2011-slice:1063 , thess-be2011-slice:1064 , thess-be2011-slice:1065 , thess-be2011-slice:1066 , thess-be2011-slice:1067 , thess-be2011-slice:1068 , thess-be2011-slice:1069 , thess-be2011-slice:107 , thess-be2011-slice:1070 , thess-be2011-slice:1071 , thess-be2011-slice:1072 , thess-be2011-slice:1073 , thess-be2011-slice:1074 , thess-be2011-slice:1075 , thess-be2011-slice:1076 , thess-be2011-slice:1077 , thess-be2011-slice:1078 , thess-be2011-slice:1079 , thess-be2011-slice:108 , thess-be2011-slice:1080 , thess-be2011-slice:1081 , thess-be2011-slice:1082 , thess-be2011-slice:1083 , thess-be2011-slice:1084 , thess-be2011-slice:1085 , thess-be2011-slice:1086 , thess-be2011-slice:1087 , thess-be2011-slice:1088 , thess-be2011-slice:1089 , thess-be2011-slice:109 , thess-be2011-slice:1090 , thess-be2011-slice:1091 , thess-be2011-slice:1092 , thess-be2011-slice:1093 , thess-be2011-slice:1094 , thess-be2011-slice:1095 , thess-be2011-slice:1096 , thess-be2011-slice:1097 , thess-be2011-slice:1098 , thess-be2011-slice:1099 , thess-be2011-slice:11 , thess-be2011-slice:110 , thess-be2011-slice:1100 , thess-be2011-slice:1101 , thess-be2011-slice:1102 , thess-be2011-slice:1103 , thess-be2011-slice:1104 , thess-be2011-slice:1105 , thess-be2011-slice:1106 , thess-be2011-slice:1107 , thess-be2011-slice:1108 , thess-be2011-slice:1109 , thess-be2011-slice:111 , thess-be2011-slice:1110 , thess-be2011-slice:1111 , thess-be2011-slice:1112 , thess-be2011-slice:1113 , thess-be2011-slice:1114 , thess-be2011-slice:1115 , thess-be2011-slice:1116 , thess-be2011-slice:1117 , thess-be2011-slice:1118 , thess-be2011-slice:1119 , thess-be2011-slice:112 , thess-be2011-slice:1120 , thess-be2011-slice:1121 , thess-be2011-slice:1122 , thess-be2011-slice:1123 , thess-be2011-slice:1124 , thess-be2011-slice:1125 , thess-be2011-slice:1126 , thess-be2011-slice:1127 , thess-be2011-slice:1128 , thess-be2011-slice:1129 , thess-be2011-slice:113 , thess-be2011-slice:1130 , thess-be2011-slice:1131 , thess-be2011-slice:1132 , thess-be2011-slice:1133 , thess-be2011-slice:1134 , thess-be2011-slice:1135 , thess-be2011-slice:1136 , thess-be2011-slice:1137 , thess-be2011-slice:1138 , thess-be2011-slice:1139 , thess-be2011-slice:114 , thess-be2011-slice:1140 , thess-be2011-slice:1141 , thess-be2011-slice:1142 , thess-be2011-slice:1143 , thess-be2011-slice:1144 , thess-be2011-slice:1145 , thess-be2011-slice:1146 , thess-be2011-slice:1147 , thess-be2011-slice:1148 , thess-be2011-slice:1149 , thess-be2011-slice:115 , thess-be2011-slice:1150 , thess-be2011-slice:1151 , thess-be2011-slice:1152 , thess-be2011-slice:1153 , thess-be2011-slice:1154 , thess-be2011-slice:1155 , thess-be2011-slice:1156 , thess-be2011-slice:1157 , thess-be2011-slice:1158 , thess-be2011-slice:1159 , thess-be2011-slice:116 , thess-be2011-slice:1160 , thess-be2011-slice:1161 , thess-be2011-slice:1162 , thess-be2011-slice:1163 , thess-be2011-slice:1164 , thess-be2011-slice:1165 , thess-be2011-slice:1166 , thess-be2011-slice:1167 , thess-be2011-slice:1168 , thess-be2011-slice:1169 , thess-be2011-slice:117 , thess-be2011-slice:1170 , thess-be2011-slice:1171 , thess-be2011-slice:1172 , thess-be2011-slice:1173 , thess-be2011-slice:1174 , thess-be2011-slice:1175 , thess-be2011-slice:1176 , thess-be2011-slice:1177 , thess-be2011-slice:1178 , thess-be2011-slice:1179 , thess-be2011-slice:118 , thess-be2011-slice:1180 , thess-be2011-slice:1181 , thess-be2011-slice:1182 , thess-be2011-slice:1183 , thess-be2011-slice:1184 , thess-be2011-slice:1185 , thess-be2011-slice:1186 , thess-be2011-slice:1187 , thess-be2011-slice:1188 , thess-be2011-slice:1189 , thess-be2011-slice:119 , thess-be2011-slice:1190 , thess-be2011-slice:1191 , thess-be2011-slice:1192 , thess-be2011-slice:1193 , thess-be2011-slice:1194 , thess-be2011-slice:1195 , thess-be2011-slice:1196 , thess-be2011-slice:1197 , thess-be2011-slice:1198 , thess-be2011-slice:1199 , thess-be2011-slice:12 , thess-be2011-slice:120 , thess-be2011-slice:1200 , thess-be2011-slice:1201 , thess-be2011-slice:1202 , thess-be2011-slice:1203 , thess-be2011-slice:1204 , thess-be2011-slice:1205 , thess-be2011-slice:1206 , thess-be2011-slice:1207 , thess-be2011-slice:1208 , thess-be2011-slice:1209 , thess-be2011-slice:121 , thess-be2011-slice:1210 , thess-be2011-slice:1211 , thess-be2011-slice:1212 , thess-be2011-slice:1213 , thess-be2011-slice:1214 , thess-be2011-slice:1215 , thess-be2011-slice:1216 , thess-be2011-slice:1217 , thess-be2011-slice:1218 , thess-be2011-slice:1219 , thess-be2011-slice:122 , thess-be2011-slice:1220 , thess-be2011-slice:1221 , thess-be2011-slice:1222 , thess-be2011-slice:1223 , thess-be2011-slice:1224 , thess-be2011-slice:1225 , thess-be2011-slice:1226 , thess-be2011-slice:1227 , thess-be2011-slice:1228 , thess-be2011-slice:1229 , thess-be2011-slice:123 , thess-be2011-slice:1230 , thess-be2011-slice:1231 , thess-be2011-slice:1232 , thess-be2011-slice:1233 , thess-be2011-slice:1234 , thess-be2011-slice:1235 , thess-be2011-slice:1236 , thess-be2011-slice:1237 , thess-be2011-slice:1238 , thess-be2011-slice:1239 , thess-be2011-slice:124 , thess-be2011-slice:1240 , thess-be2011-slice:1241 , thess-be2011-slice:1242 , thess-be2011-slice:1243 , thess-be2011-slice:1244 , thess-be2011-slice:1245 , thess-be2011-slice:1246 , thess-be2011-slice:1247 , thess-be2011-slice:1248 , thess-be2011-slice:1249 , thess-be2011-slice:125 , thess-be2011-slice:1250 , thess-be2011-slice:1251 , thess-be2011-slice:1252 , thess-be2011-slice:1253 , thess-be2011-slice:1254 , thess-be2011-slice:1255 , thess-be2011-slice:1256 , thess-be2011-slice:1257 , thess-be2011-slice:1258 , thess-be2011-slice:1259 , thess-be2011-slice:126 , thess-be2011-slice:1260 , thess-be2011-slice:1261 , thess-be2011-slice:1262 , thess-be2011-slice:1263 , thess-be2011-slice:1264 , thess-be2011-slice:1265 , thess-be2011-slice:1266 , thess-be2011-slice:1267 , thess-be2011-slice:1268 , thess-be2011-slice:1269 , thess-be2011-slice:127 , thess-be2011-slice:1270 , thess-be2011-slice:1271 , thess-be2011-slice:1272 , thess-be2011-slice:1273 , thess-be2011-slice:1274 , thess-be2011-slice:1275 , thess-be2011-slice:1276 , thess-be2011-slice:1277 , thess-be2011-slice:1278 , thess-be2011-slice:1279 , thess-be2011-slice:128 , thess-be2011-slice:1280 , thess-be2011-slice:1281 , thess-be2011-slice:1282 , thess-be2011-slice:1283 , thess-be2011-slice:1284 , thess-be2011-slice:1285 , thess-be2011-slice:1286 , thess-be2011-slice:1287 , thess-be2011-slice:1288 , thess-be2011-slice:1289 , thess-be2011-slice:129 , thess-be2011-slice:1290 , thess-be2011-slice:1291 , thess-be2011-slice:1292 , thess-be2011-slice:1293 , thess-be2011-slice:1294 , thess-be2011-slice:1295 , thess-be2011-slice:1296 , thess-be2011-slice:1297 , thess-be2011-slice:1298 , thess-be2011-slice:1299 , thess-be2011-slice:13 , thess-be2011-slice:130 , thess-be2011-slice:1300 , thess-be2011-slice:1301 , thess-be2011-slice:1302 , thess-be2011-slice:1303 , thess-be2011-slice:1304 , thess-be2011-slice:1305 , thess-be2011-slice:1306 , thess-be2011-slice:1307 , thess-be2011-slice:1308 , thess-be2011-slice:1309 , thess-be2011-slice:131 , thess-be2011-slice:1310 , thess-be2011-slice:1311 , thess-be2011-slice:1312 , thess-be2011-slice:1313 , thess-be2011-slice:1314 , thess-be2011-slice:1315 , thess-be2011-slice:1316 , thess-be2011-slice:1317 , thess-be2011-slice:1318 , thess-be2011-slice:1319 , thess-be2011-slice:132 , thess-be2011-slice:1320 , thess-be2011-slice:1321 , thess-be2011-slice:1322 , thess-be2011-slice:1323 , thess-be2011-slice:1324 , thess-be2011-slice:1325 , thess-be2011-slice:1326 , thess-be2011-slice:1327 , thess-be2011-slice:1328 , thess-be2011-slice:1329 , thess-be2011-slice:133 , thess-be2011-slice:1330 , thess-be2011-slice:1331 , thess-be2011-slice:1332 , thess-be2011-slice:1333 , thess-be2011-slice:1334 , thess-be2011-slice:1335 , thess-be2011-slice:1336 , thess-be2011-slice:1337 , thess-be2011-slice:1338 , thess-be2011-slice:1339 , thess-be2011-slice:134 , thess-be2011-slice:1340 , thess-be2011-slice:1341 , thess-be2011-slice:1342 , thess-be2011-slice:1343 , thess-be2011-slice:1344 , thess-be2011-slice:1345 , thess-be2011-slice:1346 , thess-be2011-slice:1347 , thess-be2011-slice:1348 , thess-be2011-slice:1349 , thess-be2011-slice:135 , thess-be2011-slice:1350 , thess-be2011-slice:1351 , thess-be2011-slice:1352 , thess-be2011-slice:1353 , thess-be2011-slice:1354 , thess-be2011-slice:1355 , thess-be2011-slice:1356 , thess-be2011-slice:1357 , thess-be2011-slice:1358 , thess-be2011-slice:1359 , thess-be2011-slice:136 , thess-be2011-slice:1360 , thess-be2011-slice:1361 , thess-be2011-slice:1362 , thess-be2011-slice:1363 , thess-be2011-slice:1364 , thess-be2011-slice:1365 , thess-be2011-slice:1366 , thess-be2011-slice:1367 , thess-be2011-slice:1368 , thess-be2011-slice:1369 , thess-be2011-slice:137 , thess-be2011-slice:1370 , thess-be2011-slice:1371 , thess-be2011-slice:1372 , thess-be2011-slice:1373 , thess-be2011-slice:1374 , thess-be2011-slice:1375 , thess-be2011-slice:1376 , thess-be2011-slice:1377 , thess-be2011-slice:1378 , thess-be2011-slice:1379 , thess-be2011-slice:138 , thess-be2011-slice:1380 , thess-be2011-slice:1381 , thess-be2011-slice:1382 , thess-be2011-slice:1383 , thess-be2011-slice:1384 , thess-be2011-slice:1385 , thess-be2011-slice:1386 , thess-be2011-slice:1387 , thess-be2011-slice:1388 , thess-be2011-slice:1389 , thess-be2011-slice:139 , thess-be2011-slice:1390 , thess-be2011-slice:1391 , thess-be2011-slice:1392 , thess-be2011-slice:1393 , thess-be2011-slice:1394 , thess-be2011-slice:1395 , thess-be2011-slice:1396 , thess-be2011-slice:1397 , thess-be2011-slice:1398 , thess-be2011-slice:1399 , thess-be2011-slice:14 , thess-be2011-slice:140 , thess-be2011-slice:1400 , thess-be2011-slice:1401 , thess-be2011-slice:1402 , thess-be2011-slice:1403 , thess-be2011-slice:1404 , thess-be2011-slice:1405 , thess-be2011-slice:1406 , thess-be2011-slice:1407 , thess-be2011-slice:1408 , thess-be2011-slice:1409 , thess-be2011-slice:141 , thess-be2011-slice:1410 , thess-be2011-slice:1411 , thess-be2011-slice:1412 , thess-be2011-slice:1413 , thess-be2011-slice:1414 , thess-be2011-slice:1415 , thess-be2011-slice:1416 , thess-be2011-slice:1417 , thess-be2011-slice:1418 , thess-be2011-slice:1419 , thess-be2011-slice:142 , thess-be2011-slice:1420 , thess-be2011-slice:1421 , thess-be2011-slice:1422 , thess-be2011-slice:1423 , thess-be2011-slice:1424 , thess-be2011-slice:1425 , thess-be2011-slice:1426 , thess-be2011-slice:1427 , thess-be2011-slice:1428 , thess-be2011-slice:1429 , thess-be2011-slice:143 , thess-be2011-slice:1430 , thess-be2011-slice:1431 , thess-be2011-slice:1432 , thess-be2011-slice:1433 , thess-be2011-slice:1434 , thess-be2011-slice:1435 , thess-be2011-slice:1436 , thess-be2011-slice:1437 , thess-be2011-slice:1438 , thess-be2011-slice:1439 , thess-be2011-slice:144 , thess-be2011-slice:1440 , thess-be2011-slice:1441 , thess-be2011-slice:1442 , thess-be2011-slice:1443 , thess-be2011-slice:1444 , thess-be2011-slice:1445 , thess-be2011-slice:1446 , thess-be2011-slice:1447 , thess-be2011-slice:1448 , thess-be2011-slice:1449 , thess-be2011-slice:145 , thess-be2011-slice:1450 , thess-be2011-slice:1451 , thess-be2011-slice:1452 , thess-be2011-slice:1453 , thess-be2011-slice:1454 , thess-be2011-slice:1455 , thess-be2011-slice:1456 , thess-be2011-slice:1457 , thess-be2011-slice:1458 , thess-be2011-slice:1459 , thess-be2011-slice:146 , thess-be2011-slice:1460 , thess-be2011-slice:1461 , thess-be2011-slice:1462 , thess-be2011-slice:1463 , thess-be2011-slice:1464 , thess-be2011-slice:1465 , thess-be2011-slice:1466 , thess-be2011-slice:1467 , thess-be2011-slice:1468 , thess-be2011-slice:1469 , thess-be2011-slice:147 , thess-be2011-slice:1470 , thess-be2011-slice:1471 , thess-be2011-slice:1472 , thess-be2011-slice:1473 , thess-be2011-slice:1474 , thess-be2011-slice:1475 , thess-be2011-slice:1476 , thess-be2011-slice:1477 , thess-be2011-slice:1478 , thess-be2011-slice:1479 , thess-be2011-slice:148 , thess-be2011-slice:1480 , thess-be2011-slice:1481 , thess-be2011-slice:1482 , thess-be2011-slice:1483 , thess-be2011-slice:1484 , thess-be2011-slice:1485 , thess-be2011-slice:1486 , thess-be2011-slice:1487 , thess-be2011-slice:1488 , thess-be2011-slice:1489 , thess-be2011-slice:149 , thess-be2011-slice:1490 , thess-be2011-slice:1491 , thess-be2011-slice:1492 , thess-be2011-slice:1493 , thess-be2011-slice:1494 , thess-be2011-slice:1495 , thess-be2011-slice:1496 , thess-be2011-slice:1497 , thess-be2011-slice:1498 , thess-be2011-slice:1499 , thess-be2011-slice:15 , thess-be2011-slice:150 , thess-be2011-slice:1500 , thess-be2011-slice:1501 , thess-be2011-slice:1502 , thess-be2011-slice:1503 , thess-be2011-slice:1504 , thess-be2011-slice:1505 , thess-be2011-slice:1506 , thess-be2011-slice:1507 , thess-be2011-slice:1508 , thess-be2011-slice:1509 , thess-be2011-slice:151 , thess-be2011-slice:1510 , thess-be2011-slice:1511 , thess-be2011-slice:1512 , thess-be2011-slice:1513 , thess-be2011-slice:1514 , thess-be2011-slice:1515 , thess-be2011-slice:1516 , thess-be2011-slice:1517 , thess-be2011-slice:1518 , thess-be2011-slice:1519 , thess-be2011-slice:152 , thess-be2011-slice:1520 , thess-be2011-slice:1521 , thess-be2011-slice:1522 , thess-be2011-slice:1523 , thess-be2011-slice:1524 , thess-be2011-slice:1525 , thess-be2011-slice:1526 , thess-be2011-slice:1527 , thess-be2011-slice:1528 , thess-be2011-slice:1529 , thess-be2011-slice:153 , thess-be2011-slice:1530 , thess-be2011-slice:1531 , thess-be2011-slice:1532 , thess-be2011-slice:1533 , thess-be2011-slice:1534 , thess-be2011-slice:1535 , thess-be2011-slice:1536 , thess-be2011-slice:1537 , thess-be2011-slice:1538 , thess-be2011-slice:1539 , thess-be2011-slice:154 , thess-be2011-slice:1540 , thess-be2011-slice:1541 , thess-be2011-slice:1542 , thess-be2011-slice:1543 , thess-be2011-slice:1544 , thess-be2011-slice:1545 , thess-be2011-slice:1546 , thess-be2011-slice:1547 , thess-be2011-slice:1548 , thess-be2011-slice:1549 , thess-be2011-slice:155 , thess-be2011-slice:156 , thess-be2011-slice:157 , thess-be2011-slice:158 , thess-be2011-slice:159 , thess-be2011-slice:16 , thess-be2011-slice:160 , thess-be2011-slice:161 , thess-be2011-slice:162 , thess-be2011-slice:163 , thess-be2011-slice:164 , thess-be2011-slice:165 , thess-be2011-slice:166 , thess-be2011-slice:167 , thess-be2011-slice:168 , thess-be2011-slice:169 , thess-be2011-slice:17 , thess-be2011-slice:170 , thess-be2011-slice:171 , thess-be2011-slice:172 , thess-be2011-slice:173 , thess-be2011-slice:174 , thess-be2011-slice:175 , thess-be2011-slice:176 , thess-be2011-slice:177 , thess-be2011-slice:178 , thess-be2011-slice:179 , thess-be2011-slice:18 , thess-be2011-slice:180 , thess-be2011-slice:181 , thess-be2011-slice:182 , thess-be2011-slice:183 , thess-be2011-slice:184 , thess-be2011-slice:185 , thess-be2011-slice:186 , thess-be2011-slice:187 , thess-be2011-slice:188 , thess-be2011-slice:189 , thess-be2011-slice:19 , thess-be2011-slice:190 , thess-be2011-slice:191 , thess-be2011-slice:192 , thess-be2011-slice:193 , thess-be2011-slice:194 , thess-be2011-slice:195 , thess-be2011-slice:196 , thess-be2011-slice:197 , thess-be2011-slice:198 , thess-be2011-slice:199 , thess-be2011-slice:2 , thess-be2011-slice:20 , thess-be2011-slice:200 , thess-be2011-slice:201 , thess-be2011-slice:202 , thess-be2011-slice:203 , thess-be2011-slice:204 , thess-be2011-slice:205 , thess-be2011-slice:206 , thess-be2011-slice:207 , thess-be2011-slice:208 , thess-be2011-slice:209 , thess-be2011-slice:21 , thess-be2011-slice:210 , thess-be2011-slice:211 , thess-be2011-slice:212 , thess-be2011-slice:213 , thess-be2011-slice:214 , thess-be2011-slice:215 , thess-be2011-slice:216 , thess-be2011-slice:217 , thess-be2011-slice:218 , thess-be2011-slice:219 , thess-be2011-slice:22 , thess-be2011-slice:220 , thess-be2011-slice:221 , thess-be2011-slice:222 , thess-be2011-slice:223 , thess-be2011-slice:224 , thess-be2011-slice:225 , thess-be2011-slice:226 , thess-be2011-slice:227 , thess-be2011-slice:228 , thess-be2011-slice:229 , thess-be2011-slice:23 , thess-be2011-slice:230 , thess-be2011-slice:231 , thess-be2011-slice:232 , thess-be2011-slice:233 , thess-be2011-slice:234 , thess-be2011-slice:235 , thess-be2011-slice:236 , thess-be2011-slice:237 , thess-be2011-slice:238 , thess-be2011-slice:239 , thess-be2011-slice:24 , thess-be2011-slice:240 , thess-be2011-slice:241 , thess-be2011-slice:242 , thess-be2011-slice:243 , thess-be2011-slice:244 , thess-be2011-slice:245 , thess-be2011-slice:246 , thess-be2011-slice:247 , thess-be2011-slice:248 , thess-be2011-slice:249 , thess-be2011-slice:25 , thess-be2011-slice:250 , thess-be2011-slice:251 , thess-be2011-slice:252 , thess-be2011-slice:253 , thess-be2011-slice:254 , thess-be2011-slice:255 , thess-be2011-slice:256 , thess-be2011-slice:257 , thess-be2011-slice:258 , thess-be2011-slice:259 , thess-be2011-slice:26 , thess-be2011-slice:260 , thess-be2011-slice:261 , thess-be2011-slice:262 , thess-be2011-slice:263 , thess-be2011-slice:264 , thess-be2011-slice:265 , thess-be2011-slice:266 , thess-be2011-slice:267 , thess-be2011-slice:268 , thess-be2011-slice:269 , thess-be2011-slice:27 , thess-be2011-slice:270 , thess-be2011-slice:271 , thess-be2011-slice:272 , thess-be2011-slice:273 , thess-be2011-slice:274 , thess-be2011-slice:275 , thess-be2011-slice:276 , thess-be2011-slice:277 , thess-be2011-slice:278 , thess-be2011-slice:279 , thess-be2011-slice:28 , thess-be2011-slice:280 , thess-be2011-slice:281 , thess-be2011-slice:282 , thess-be2011-slice:283 , thess-be2011-slice:284 , thess-be2011-slice:285 , thess-be2011-slice:286 , thess-be2011-slice:287 , thess-be2011-slice:288 , thess-be2011-slice:289 , thess-be2011-slice:29 , thess-be2011-slice:290 , thess-be2011-slice:291 , thess-be2011-slice:292 , thess-be2011-slice:293 , thess-be2011-slice:294 , thess-be2011-slice:295 , thess-be2011-slice:296 , thess-be2011-slice:297 , thess-be2011-slice:298 , thess-be2011-slice:299 , thess-be2011-slice:3 , thess-be2011-slice:30 , thess-be2011-slice:300 , thess-be2011-slice:301 , thess-be2011-slice:302 , thess-be2011-slice:303 , thess-be2011-slice:304 , thess-be2011-slice:305 , thess-be2011-slice:306 , thess-be2011-slice:307 , thess-be2011-slice:308 , thess-be2011-slice:309 , thess-be2011-slice:31 , thess-be2011-slice:310 , thess-be2011-slice:311 , thess-be2011-slice:312 , thess-be2011-slice:313 , thess-be2011-slice:314 , thess-be2011-slice:315 , thess-be2011-slice:316 , thess-be2011-slice:317 , thess-be2011-slice:318 , thess-be2011-slice:319 , thess-be2011-slice:32 , thess-be2011-slice:320 , thess-be2011-slice:321 , thess-be2011-slice:322 , thess-be2011-slice:323 , thess-be2011-slice:324 , thess-be2011-slice:325 , thess-be2011-slice:326 , thess-be2011-slice:327 , thess-be2011-slice:328 , thess-be2011-slice:329 , thess-be2011-slice:33 , thess-be2011-slice:330 , thess-be2011-slice:331 , thess-be2011-slice:332 , thess-be2011-slice:333 , thess-be2011-slice:334 , thess-be2011-slice:335 , thess-be2011-slice:336 , thess-be2011-slice:337 , thess-be2011-slice:338 , thess-be2011-slice:339 , thess-be2011-slice:34 , thess-be2011-slice:340 , thess-be2011-slice:341 , thess-be2011-slice:342 , thess-be2011-slice:343 , thess-be2011-slice:344 , thess-be2011-slice:345 , thess-be2011-slice:346 , thess-be2011-slice:347 , thess-be2011-slice:348 , thess-be2011-slice:349 , thess-be2011-slice:35 , thess-be2011-slice:350 , thess-be2011-slice:351 , thess-be2011-slice:352 , thess-be2011-slice:353 , thess-be2011-slice:354 , thess-be2011-slice:355 , thess-be2011-slice:356 , thess-be2011-slice:357 , thess-be2011-slice:358 , thess-be2011-slice:359 , thess-be2011-slice:36 , thess-be2011-slice:360 , thess-be2011-slice:361 , thess-be2011-slice:362 , thess-be2011-slice:363 , thess-be2011-slice:364 , thess-be2011-slice:365 , thess-be2011-slice:366 , thess-be2011-slice:367 , thess-be2011-slice:368 , thess-be2011-slice:369 , thess-be2011-slice:37 , thess-be2011-slice:370 , thess-be2011-slice:371 , thess-be2011-slice:372 , thess-be2011-slice:373 , thess-be2011-slice:374 , thess-be2011-slice:375 , thess-be2011-slice:376 , thess-be2011-slice:377 , thess-be2011-slice:378 , thess-be2011-slice:379 , thess-be2011-slice:38 , thess-be2011-slice:380 , thess-be2011-slice:381 , thess-be2011-slice:382 , thess-be2011-slice:383 , thess-be2011-slice:384 , thess-be2011-slice:385 , thess-be2011-slice:386 , thess-be2011-slice:387 , thess-be2011-slice:388 , thess-be2011-slice:389 , thess-be2011-slice:39 , thess-be2011-slice:390 , thess-be2011-slice:391 , thess-be2011-slice:392 , thess-be2011-slice:393 , thess-be2011-slice:394 , thess-be2011-slice:395 , thess-be2011-slice:396 , thess-be2011-slice:397 , thess-be2011-slice:398 , thess-be2011-slice:399 , thess-be2011-slice:4 , thess-be2011-slice:40 , thess-be2011-slice:400 , thess-be2011-slice:401 , thess-be2011-slice:402 , thess-be2011-slice:403 , thess-be2011-slice:404 , thess-be2011-slice:405 , thess-be2011-slice:406 , thess-be2011-slice:407 , thess-be2011-slice:408 , thess-be2011-slice:409 , thess-be2011-slice:41 , thess-be2011-slice:410 , thess-be2011-slice:411 , thess-be2011-slice:412 , thess-be2011-slice:413 , thess-be2011-slice:414 , thess-be2011-slice:415 , thess-be2011-slice:416 , thess-be2011-slice:417 , thess-be2011-slice:418 , thess-be2011-slice:419 , thess-be2011-slice:42 , thess-be2011-slice:420 , thess-be2011-slice:421 , thess-be2011-slice:422 , thess-be2011-slice:423 , thess-be2011-slice:424 , thess-be2011-slice:425 , thess-be2011-slice:426 , thess-be2011-slice:427 , thess-be2011-slice:428 , thess-be2011-slice:429 , thess-be2011-slice:43 , thess-be2011-slice:430 , thess-be2011-slice:431 , thess-be2011-slice:432 , thess-be2011-slice:433 , thess-be2011-slice:434 , thess-be2011-slice:435 , thess-be2011-slice:436 , thess-be2011-slice:437 , thess-be2011-slice:438 , thess-be2011-slice:439 , thess-be2011-slice:44 , thess-be2011-slice:440 , thess-be2011-slice:441 , thess-be2011-slice:442 , thess-be2011-slice:443 , thess-be2011-slice:444 , thess-be2011-slice:445 , thess-be2011-slice:446 , thess-be2011-slice:447 , thess-be2011-slice:448 , thess-be2011-slice:449 , thess-be2011-slice:45 , thess-be2011-slice:450 , thess-be2011-slice:451 , thess-be2011-slice:452 , thess-be2011-slice:453 , thess-be2011-slice:454 , thess-be2011-slice:455 , thess-be2011-slice:456 , thess-be2011-slice:457 , thess-be2011-slice:458 , thess-be2011-slice:459 , thess-be2011-slice:46 , thess-be2011-slice:460 , thess-be2011-slice:461 , thess-be2011-slice:462 , thess-be2011-slice:463 , thess-be2011-slice:464 , thess-be2011-slice:465 , thess-be2011-slice:466 , thess-be2011-slice:467 , thess-be2011-slice:468 , thess-be2011-slice:469 , thess-be2011-slice:47 , thess-be2011-slice:470 , thess-be2011-slice:471 , thess-be2011-slice:472 , thess-be2011-slice:473 , thess-be2011-slice:474 , thess-be2011-slice:475 , thess-be2011-slice:476 , thess-be2011-slice:477 , thess-be2011-slice:478 , thess-be2011-slice:479 , thess-be2011-slice:48 , thess-be2011-slice:480 , thess-be2011-slice:481 , thess-be2011-slice:482 , thess-be2011-slice:483 , thess-be2011-slice:484 , thess-be2011-slice:485 , thess-be2011-slice:486 , thess-be2011-slice:487 , thess-be2011-slice:488 , thess-be2011-slice:489 , thess-be2011-slice:49 , thess-be2011-slice:490 , thess-be2011-slice:491 , thess-be2011-slice:492 , thess-be2011-slice:493 , thess-be2011-slice:494 , thess-be2011-slice:495 , thess-be2011-slice:496 , thess-be2011-slice:497 , thess-be2011-slice:498 , thess-be2011-slice:499 , thess-be2011-slice:5 , thess-be2011-slice:50 , thess-be2011-slice:500 , thess-be2011-slice:501 , thess-be2011-slice:502 , thess-be2011-slice:503 , thess-be2011-slice:504 , thess-be2011-slice:505 , thess-be2011-slice:506 , thess-be2011-slice:507 , thess-be2011-slice:508 , thess-be2011-slice:509 , thess-be2011-slice:51 , thess-be2011-slice:510 , thess-be2011-slice:511 , thess-be2011-slice:512 , thess-be2011-slice:513 , thess-be2011-slice:514 , thess-be2011-slice:515 , thess-be2011-slice:516 , thess-be2011-slice:517 , thess-be2011-slice:518 , thess-be2011-slice:519 , thess-be2011-slice:52 , thess-be2011-slice:520 , thess-be2011-slice:521 , thess-be2011-slice:522 , thess-be2011-slice:523 , thess-be2011-slice:524 , thess-be2011-slice:525 , thess-be2011-slice:526 , thess-be2011-slice:527 , thess-be2011-slice:528 , thess-be2011-slice:529 , thess-be2011-slice:53 , thess-be2011-slice:530 , thess-be2011-slice:531 , thess-be2011-slice:532 , thess-be2011-slice:533 , thess-be2011-slice:534 , thess-be2011-slice:535 , thess-be2011-slice:536 , thess-be2011-slice:537 , thess-be2011-slice:538 , thess-be2011-slice:539 , thess-be2011-slice:54 , thess-be2011-slice:540 , thess-be2011-slice:541 , thess-be2011-slice:542 , thess-be2011-slice:543 , thess-be2011-slice:544 , thess-be2011-slice:545 , thess-be2011-slice:546 , thess-be2011-slice:547 , thess-be2011-slice:548 , thess-be2011-slice:549 , thess-be2011-slice:55 , thess-be2011-slice:550 , thess-be2011-slice:551 , thess-be2011-slice:552 , thess-be2011-slice:553 , thess-be2011-slice:554 , thess-be2011-slice:555 , thess-be2011-slice:556 , thess-be2011-slice:557 , thess-be2011-slice:558 , thess-be2011-slice:559 , thess-be2011-slice:56 , thess-be2011-slice:560 , thess-be2011-slice:561 , thess-be2011-slice:562 , thess-be2011-slice:563 , thess-be2011-slice:564 , thess-be2011-slice:565 , thess-be2011-slice:566 , thess-be2011-slice:567 , thess-be2011-slice:568 , thess-be2011-slice:569 , thess-be2011-slice:57 , thess-be2011-slice:570 , thess-be2011-slice:571 , thess-be2011-slice:572 , thess-be2011-slice:573 , thess-be2011-slice:574 , thess-be2011-slice:575 , thess-be2011-slice:576 , thess-be2011-slice:577 , thess-be2011-slice:578 , thess-be2011-slice:579 , thess-be2011-slice:58 , thess-be2011-slice:580 , thess-be2011-slice:581 , thess-be2011-slice:582 , thess-be2011-slice:583 , thess-be2011-slice:584 , thess-be2011-slice:585 , thess-be2011-slice:586 , thess-be2011-slice:587 , thess-be2011-slice:588 , thess-be2011-slice:589 , thess-be2011-slice:59 , thess-be2011-slice:590 , thess-be2011-slice:591 , thess-be2011-slice:592 , thess-be2011-slice:593 , thess-be2011-slice:594 , thess-be2011-slice:595 , thess-be2011-slice:596 , thess-be2011-slice:597 , thess-be2011-slice:598 , thess-be2011-slice:599 , thess-be2011-slice:6 , thess-be2011-slice:60 , thess-be2011-slice:600 , thess-be2011-slice:601 , thess-be2011-slice:602 , thess-be2011-slice:603 , thess-be2011-slice:604 , thess-be2011-slice:605 , thess-be2011-slice:606 , thess-be2011-slice:607 , thess-be2011-slice:608 , thess-be2011-slice:609 , thess-be2011-slice:61 , thess-be2011-slice:610 , thess-be2011-slice:611 , thess-be2011-slice:612 , thess-be2011-slice:613 , thess-be2011-slice:614 , thess-be2011-slice:615 , thess-be2011-slice:616 , thess-be2011-slice:617 , thess-be2011-slice:618 , thess-be2011-slice:619 , thess-be2011-slice:62 , thess-be2011-slice:620 , thess-be2011-slice:621 , thess-be2011-slice:622 , thess-be2011-slice:623 , thess-be2011-slice:624 , thess-be2011-slice:625 , thess-be2011-slice:626 , thess-be2011-slice:627 , thess-be2011-slice:628 , thess-be2011-slice:629 , thess-be2011-slice:63 , thess-be2011-slice:630 , thess-be2011-slice:631 , thess-be2011-slice:632 , thess-be2011-slice:633 , thess-be2011-slice:634 , thess-be2011-slice:635 , thess-be2011-slice:636 , thess-be2011-slice:637 , thess-be2011-slice:638 , thess-be2011-slice:639 , thess-be2011-slice:64 , thess-be2011-slice:640 , thess-be2011-slice:641 , thess-be2011-slice:642 , thess-be2011-slice:643 , thess-be2011-slice:644 , thess-be2011-slice:645 , thess-be2011-slice:646 , thess-be2011-slice:647 , thess-be2011-slice:648 , thess-be2011-slice:649 , thess-be2011-slice:65 , thess-be2011-slice:650 , thess-be2011-slice:651 , thess-be2011-slice:652 , thess-be2011-slice:653 , thess-be2011-slice:654 , thess-be2011-slice:655 , thess-be2011-slice:656 , thess-be2011-slice:657 , thess-be2011-slice:658 , thess-be2011-slice:659 , thess-be2011-slice:66 , thess-be2011-slice:660 , thess-be2011-slice:661 , thess-be2011-slice:662 , thess-be2011-slice:663 , thess-be2011-slice:664 , thess-be2011-slice:665 , thess-be2011-slice:666 , thess-be2011-slice:667 , thess-be2011-slice:668 , thess-be2011-slice:669 , thess-be2011-slice:67 , thess-be2011-slice:670 , thess-be2011-slice:671 , thess-be2011-slice:672 , thess-be2011-slice:673 , thess-be2011-slice:674 , thess-be2011-slice:675 , thess-be2011-slice:676 , thess-be2011-slice:677 , thess-be2011-slice:678 , thess-be2011-slice:679 , thess-be2011-slice:68 , thess-be2011-slice:680 , thess-be2011-slice:681 , thess-be2011-slice:682 , thess-be2011-slice:683 , thess-be2011-slice:684 , thess-be2011-slice:685 , thess-be2011-slice:686 , thess-be2011-slice:687 , thess-be2011-slice:688 , thess-be2011-slice:689 , thess-be2011-slice:69 , thess-be2011-slice:690 , thess-be2011-slice:691 , thess-be2011-slice:692 , thess-be2011-slice:693 , thess-be2011-slice:694 , thess-be2011-slice:695 , thess-be2011-slice:696 , thess-be2011-slice:697 , thess-be2011-slice:698 , thess-be2011-slice:699 , thess-be2011-slice:7 , thess-be2011-slice:70 , thess-be2011-slice:700 , thess-be2011-slice:701 , thess-be2011-slice:702 , thess-be2011-slice:703 , thess-be2011-slice:704 , thess-be2011-slice:705 , thess-be2011-slice:706 , thess-be2011-slice:707 , thess-be2011-slice:708 , thess-be2011-slice:709 , thess-be2011-slice:71 , thess-be2011-slice:710 , thess-be2011-slice:711 , thess-be2011-slice:712 , thess-be2011-slice:713 , thess-be2011-slice:714 , thess-be2011-slice:715 , thess-be2011-slice:716 , thess-be2011-slice:717 , thess-be2011-slice:718 , thess-be2011-slice:719 , thess-be2011-slice:72 , thess-be2011-slice:720 , thess-be2011-slice:721 , thess-be2011-slice:722 , thess-be2011-slice:723 , thess-be2011-slice:724 , thess-be2011-slice:725 , thess-be2011-slice:726 , thess-be2011-slice:727 , thess-be2011-slice:728 , thess-be2011-slice:729 , thess-be2011-slice:73 , thess-be2011-slice:730 , thess-be2011-slice:731 , thess-be2011-slice:732 , thess-be2011-slice:733 , thess-be2011-slice:734 , thess-be2011-slice:735 , thess-be2011-slice:736 , thess-be2011-slice:737 , thess-be2011-slice:738 , thess-be2011-slice:739 , thess-be2011-slice:74 , thess-be2011-slice:740 , thess-be2011-slice:741 , thess-be2011-slice:742 , thess-be2011-slice:743 , thess-be2011-slice:744 , thess-be2011-slice:745 , thess-be2011-slice:746 , thess-be2011-slice:747 , thess-be2011-slice:748 , thess-be2011-slice:749 , thess-be2011-slice:75 , thess-be2011-slice:750 , thess-be2011-slice:751 , thess-be2011-slice:752 , thess-be2011-slice:753 , thess-be2011-slice:754 , thess-be2011-slice:755 , thess-be2011-slice:756 , thess-be2011-slice:757 , thess-be2011-slice:758 , thess-be2011-slice:759 , thess-be2011-slice:76 , thess-be2011-slice:760 , thess-be2011-slice:761 , thess-be2011-slice:762 , thess-be2011-slice:763 , thess-be2011-slice:764 , thess-be2011-slice:765 , thess-be2011-slice:766 , thess-be2011-slice:767 , thess-be2011-slice:768 , thess-be2011-slice:769 , thess-be2011-slice:77 , thess-be2011-slice:770 , thess-be2011-slice:771 , thess-be2011-slice:772 , thess-be2011-slice:773 , thess-be2011-slice:774 , thess-be2011-slice:775 , thess-be2011-slice:776 , thess-be2011-slice:777 , thess-be2011-slice:778 , thess-be2011-slice:779 , thess-be2011-slice:78 , thess-be2011-slice:780 , thess-be2011-slice:781 , thess-be2011-slice:782 , thess-be2011-slice:783 , thess-be2011-slice:784 , thess-be2011-slice:785 , thess-be2011-slice:786 , thess-be2011-slice:787 , thess-be2011-slice:788 , thess-be2011-slice:789 , thess-be2011-slice:79 , thess-be2011-slice:790 , thess-be2011-slice:791 , thess-be2011-slice:792 , thess-be2011-slice:793 , thess-be2011-slice:794 , thess-be2011-slice:795 , thess-be2011-slice:796 , thess-be2011-slice:797 , thess-be2011-slice:798 , thess-be2011-slice:799 , thess-be2011-slice:8 , thess-be2011-slice:80 , thess-be2011-slice:800 , thess-be2011-slice:801 , thess-be2011-slice:802 , thess-be2011-slice:803 , thess-be2011-slice:804 , thess-be2011-slice:805 , thess-be2011-slice:806 , thess-be2011-slice:807 , thess-be2011-slice:808 , thess-be2011-slice:809 , thess-be2011-slice:81 , thess-be2011-slice:810 , thess-be2011-slice:811 , thess-be2011-slice:812 , thess-be2011-slice:813 , thess-be2011-slice:814 , thess-be2011-slice:815 , thess-be2011-slice:816 , thess-be2011-slice:817 , thess-be2011-slice:818 , thess-be2011-slice:819 , thess-be2011-slice:82 , thess-be2011-slice:820 , thess-be2011-slice:821 , thess-be2011-slice:822 , thess-be2011-slice:823 , thess-be2011-slice:824 , thess-be2011-slice:825 , thess-be2011-slice:826 , thess-be2011-slice:827 , thess-be2011-slice:828 , thess-be2011-slice:829 , thess-be2011-slice:83 , thess-be2011-slice:830 , thess-be2011-slice:831 , thess-be2011-slice:832 , thess-be2011-slice:833 , thess-be2011-slice:834 , thess-be2011-slice:835 , thess-be2011-slice:836 , thess-be2011-slice:837 , thess-be2011-slice:838 , thess-be2011-slice:839 , thess-be2011-slice:84 , thess-be2011-slice:840 , thess-be2011-slice:841 , thess-be2011-slice:842 , thess-be2011-slice:843 , thess-be2011-slice:844 , thess-be2011-slice:845 , thess-be2011-slice:846 , thess-be2011-slice:847 , thess-be2011-slice:848 , thess-be2011-slice:849 , thess-be2011-slice:85 , thess-be2011-slice:850 , thess-be2011-slice:851 , thess-be2011-slice:852 , thess-be2011-slice:853 , thess-be2011-slice:854 , thess-be2011-slice:855 , thess-be2011-slice:856 , thess-be2011-slice:857 , thess-be2011-slice:858 , thess-be2011-slice:859 , thess-be2011-slice:86 , thess-be2011-slice:860 , thess-be2011-slice:861 , thess-be2011-slice:862 , thess-be2011-slice:863 , thess-be2011-slice:864 , thess-be2011-slice:865 , thess-be2011-slice:866 , thess-be2011-slice:867 , thess-be2011-slice:868 , thess-be2011-slice:869 , thess-be2011-slice:87 , thess-be2011-slice:870 , thess-be2011-slice:871 , thess-be2011-slice:872 , thess-be2011-slice:873 , thess-be2011-slice:874 , thess-be2011-slice:875 , thess-be2011-slice:876 , thess-be2011-slice:877 , thess-be2011-slice:878 , thess-be2011-slice:879 , thess-be2011-slice:88 , thess-be2011-slice:880 , thess-be2011-slice:881 , thess-be2011-slice:882 , thess-be2011-slice:883 , thess-be2011-slice:884 , thess-be2011-slice:885 , thess-be2011-slice:886 , thess-be2011-slice:887 , thess-be2011-slice:888 , thess-be2011-slice:889 , thess-be2011-slice:89 , thess-be2011-slice:890 , thess-be2011-slice:891 , thess-be2011-slice:892 , thess-be2011-slice:893 , thess-be2011-slice:894 , thess-be2011-slice:895 , thess-be2011-slice:896 , thess-be2011-slice:897 , thess-be2011-slice:898 , thess-be2011-slice:899 , thess-be2011-slice:9 , thess-be2011-slice:90 , thess-be2011-slice:900 , thess-be2011-slice:901 , thess-be2011-slice:902 , thess-be2011-slice:903 , thess-be2011-slice:904 , thess-be2011-slice:905 , thess-be2011-slice:906 , thess-be2011-slice:907 , thess-be2011-slice:908 , thess-be2011-slice:909 , thess-be2011-slice:91 , thess-be2011-slice:910 , thess-be2011-slice:911 , thess-be2011-slice:912 , thess-be2011-slice:913 , thess-be2011-slice:914 , thess-be2011-slice:915 , thess-be2011-slice:916 , thess-be2011-slice:917 , thess-be2011-slice:918 , thess-be2011-slice:919 , thess-be2011-slice:92 , thess-be2011-slice:920 , thess-be2011-slice:921 , thess-be2011-slice:922 , thess-be2011-slice:923 , thess-be2011-slice:924 , thess-be2011-slice:925 , thess-be2011-slice:926 , thess-be2011-slice:927 , thess-be2011-slice:928 , thess-be2011-slice:929 , thess-be2011-slice:93 , thess-be2011-slice:930 , thess-be2011-slice:931 , thess-be2011-slice:932 , thess-be2011-slice:933 , thess-be2011-slice:934 , thess-be2011-slice:935 , thess-be2011-slice:936 , thess-be2011-slice:937 , thess-be2011-slice:938 , thess-be2011-slice:939 , thess-be2011-slice:94 , thess-be2011-slice:940 , thess-be2011-slice:941 , thess-be2011-slice:942 , thess-be2011-slice:943 , thess-be2011-slice:944 , thess-be2011-slice:945 , thess-be2011-slice:946 , thess-be2011-slice:947 , thess-be2011-slice:948 , thess-be2011-slice:949 , thess-be2011-slice:95 , thess-be2011-slice:950 , thess-be2011-slice:951 , thess-be2011-slice:952 , thess-be2011-slice:953 , thess-be2011-slice:954 , thess-be2011-slice:955 , thess-be2011-slice:956 , thess-be2011-slice:957 , thess-be2011-slice:958 , thess-be2011-slice:959 , thess-be2011-slice:96 , thess-be2011-slice:960 , thess-be2011-slice:961 , thess-be2011-slice:962 , thess-be2011-slice:963 , thess-be2011-slice:964 , thess-be2011-slice:965 , thess-be2011-slice:966 , thess-be2011-slice:967 , thess-be2011-slice:968 , thess-be2011-slice:969 , thess-be2011-slice:97 , thess-be2011-slice:970 , thess-be2011-slice:971 , thess-be2011-slice:972 , thess-be2011-slice:973 , thess-be2011-slice:974 , thess-be2011-slice:975 , thess-be2011-slice:976 , thess-be2011-slice:977 , thess-be2011-slice:978 , thess-be2011-slice:979 , thess-be2011-slice:98 , thess-be2011-slice:980 , thess-be2011-slice:981 , thess-be2011-slice:982 , thess-be2011-slice:983 , thess-be2011-slice:984 , thess-be2011-slice:985 , thess-be2011-slice:986 , thess-be2011-slice:987 , thess-be2011-slice:988 , thess-be2011-slice:989 , thess-be2011-slice:99 , thess-be2011-slice:990 , thess-be2011-slice:991 , thess-be2011-slice:992 , thess-be2011-slice:993 , thess-be2011-slice:994 , thess-be2011-slice:995 , thess-be2011-slice:996 , thess-be2011-slice:997 , thess-be2011-slice:998 , thess-be2011-slice:999 .

<http://openbudgets.eu> a foaf:Agent .

<http://data.openbudgets.eu/resource/dataset/budget-thessaloniki-expenditure-2011/contactPoint> a <http://www.w3.org/2006/vcard/ns#VCard> ;
    <http://www.w3.org/2006/vcard/ns#fn> "Open Knowledge Foundation Greek Chapter" ;
    <http://www.w3.org/2006/vcard/ns#hasEmail> "info@okfn.gr" .

_:node1aj2g43bfx4084 a foaf:Organization ;
    foaf:name "OKF Greece"@en ;
    foaf:homepage <http://www.okfn.gr> ;
    foaf:member _:node1aj2g43bfx4085 , _:node1aj2g43bfx4086 .

_:node1aj2g43bfx4085 a foaf:Person ;
    foaf:name "Charalampos Bratsas" .

_:node1aj2g43bfx4086 a foaf:Person ;
    foaf:name "Sotiris Karampatakis" .
```

Q1: The problem is that these are contained within the same named graph, also distribution metadata is missing as this specification was only recently introduced. Should we seperate them and include the distribution metadata?

This would require manual editing of all to upload datasets pipelines, which in our case the count is over 80.

Moreover, uploading to Fuseki and public dumps via LP  would require also manual editing of all pipelines. 

In  my understanding those datasets are to be uploaded to test data mining and other under development tasks.

Q2: If we need to be ready on prototype deadline we can simply bulk upload all datasets "as is". 

Editing of the pipelines could be performed on a later stage, where D1.5 will be on a stable form. I believe we should first decide "officially" what metadata to include and what not, in which format should be available ( we use RDF/XML and Turtle, you proposed above TRIG), resolve issues with the data model (if any) and then edit the pipelines, for the last time.

**Comment by jindrichmynarz** [comment]
> Q1: The problem is that these are contained within the same named graph, also distribution metadata is missing as this specification was only recently introduced. Should we separate them and include the distribution metadata?

Yes. See [here](https://github.com/openbudgets/platform/issues/13#issuecomment-234502670).

> This would require manual editing of all to upload datasets pipelines, which in our case the count is over 80.

As I argued above, the effort invested in metadata might outweigh the value of metadata in the OpenBudgets.eu project. Unlike the requirements for dataset metadata for visualization, I current don't see a way in which distribution metadata will be used. @jakubklimek argued above that adding distribution metadata is a matter of "few clicks", however, the situation is different for a large number of datasets. Hence, I think we should first establish how distribution metadata is to be used in the project.

> Moreover, uploading to Fuseki and public dumps via LP would require also manual editing of all pipelines.

Since LP-ETL pipelines are represented in RDF, you can write a SPARQL Update to change configuration of multiple pipelines at once. However, with deserializing JSON-LD and reserializing it back it may turn out to require more effort than manual editing.

> Q2: If we need to be ready on prototype deadline we can simply bulk upload all datasets "as is".

If we still stick with the requirements to dataset metadata, changes to the pipelines producing the data will be necessary anyway, so I don't think we'd save much effort by uploading dataset dumps first.

> Editing of the pipelines could be performed on a later stage, where D1.5 will be on a stable form. 

The data model is very stable. But I agree with the motivation that it would be best to wait for the D1.5 to be finalized.

**Comment by skarampatakis** [comment]
> The data model is very stable. But I agree with the motivation that it would be best to wait for the D1.5 to be finalized.

I was referring to general guidelines that will be present I believe in D 1.5, such as dataset and metadata IRI conventions, mandatory and optional properties on metadata, serialization format(?), and stuff like that.  

> If we still stick with the requirements to dataset metadata, changes to the pipelines producing the data will be necessary anyway, so I don't think we'd save much effort by uploading dataset dumps first.

Sure, but if other tasks are blocked by dataset uploading task, should be postponed? In the case of our datasets, most of the basic properties suggested are already present. I don't believe that this is of concern by any other task. We just save the effort of double editing and re-running the pipelines, task we have done already a couple of times.

**Comment by jindrichmynarz** [comment]
> I was referring to general guidelines that will be present I believe in D 1.5, such as dataset and metadata IRI conventions, mandatory and optional properties on metadata, serialization format(?), and stuff like that.

I see. This is mostly already scattered in various GitHub comments.
- Dataset IRI is the same as the `qb:DataSet` instance IRI, which is the same as the dataset's named graph IRI.
- Metadata IRI is created by appending `/metadata` to the dataset IRI.
- Mandatory metadata properties are described above in [this comment](https://github.com/openbudgets/platform/issues/13#issuecomment-234490081). All other metadata properties are optional.
- Serialization format: TriG (unless there are objections against it)

All of this will (and more) be added to D1.5.

> Sure, but if other tasks are blocked by dataset uploading task, should be postponed? In the case of our datasets, most of the basic properties suggested are already present. I don't believe that this is of concern by any other task. We just save the effort of double editing and re-running the pipelines, task we have done already a couple of times.

Then I would simply upload the pipelines as they are and adjust them later.

**Comment by skarampatakis** [comment]
> I see. This is mostly already scattered in various GitHub comments.
> 
> All of this will (and more) be added to D1.5.

That's the reason I insist on concrete, final decision that should be taken before re-editing the pipelines. And the proper place to be is the final version of D 1.5. 

> Then I would simply upload the pipelines as they are and adjust them later.

If we are not going to re-run the pipelines for now why to upload them? While this is a pretty simple task by a shell script, I believe we can just bulk upload the datasets on Fuseki and serialized dataset files in whatever format on /dump folder. We can upload final version of the pipelines after D 1.5.

**Comment by jindrichmynarz** [comment]
Actually, I think uploading all pipelines would be an overkill at this moment. What we need to tell first is if we can re-run the pipelines on a different server. So I would start with 1-2 pipelines and see if they work. Then we can adjust them, see if they match our requirements, and only then migrate all the remaining pipelines.

**Comment by skarampatakis** [comment]
Now we are on the same page. Thank you!

**Comment by pwalsh** [comment]
@badmotor @jakubklimek @jindrichmynarz  I'm closing this. It must be out of date, and any specific issues can be raised with distinct issues.
